diff --git a/configure.in b/configure.in
index 6cc3569..80aeec8 100644
--- a/configure.in
+++ b/configure.in
@@ -340,6 +340,9 @@ AC_ARG_ENABLE(autodepend,
  		 found])],
 [ac_enable_autodepend="${enableval}"],[ac_enable_autodepend=yes])
 
+AC_ARG_WITH([ns-app-dir],dnl
+[AS_HELP_STRING([--with-ns-app-dir=DIR],[path to NeXTstep application directory])])
+
 #### Make srcdir absolute, if it isn't already.  It's important to
 #### avoid running the path through pwd unnecessarily, since pwd can
 #### give you automounter prefixes, which can go away.  We do all this
@@ -1460,7 +1463,11 @@ dnl trailing "/" to it, so now we do it here.
 if test "${with_ns}" != no; then
   if test "${opsys}" = darwin; then
      NS_IMPL_COCOA=yes
-     ns_appdir=`pwd`/nextstep/Emacs.app
+     if test "X${with_ns_app_dir}" != X; then
+        ns_appdir="${with_ns_app_dir}"
+     else
+        ns_appdir=`pwd`/nextstep/Emacs.app
+     fi
      ns_appbindir=${ns_appdir}/Contents/MacOS/
      ns_appresdir=${ns_appdir}/Contents/Resources
      ns_appsrc=${srcdir}/nextstep/Cocoa/Emacs.base
