--- a/Makefile.in
+++ b/Makefile.in
@@ -475,14 +475,16 @@
 	rm -f $(DESTDIR)${bindir}/$(EMACS)
 	-ln $(DESTDIR)${bindir}/$(EMACSFULL) $(DESTDIR)${bindir}/$(EMACS)
 	if test "${ns_appresdir}" != ""; then \
-	  ( cd ${ns_appresdir} ; \
-	    if test -d share/emacs ; then dir=share/emacs/*/*; $(MV_DIRS); fi;\
-	    if test -d share/info ; then dir=share/info; $(MV_DIRS) ; fi ; \
-	    rm -fr share ) ; \
-	  ( cd ${ns_appbindir}libexec ; dir=emacs/*/*/* ; $(MV_DIRS); \
-	    rm -fr emacs ) ; \
-	  ( cd ${ns_appbindir}bin ; rm -f emacs emacs-24* ; \
-	    ln -sf ../libexec/* .) ; \
+	  ( cd $(DESTDIR)${ns_appresdir} && \
+	    ( if test -d share/emacs ; then dir=share/emacs/*/*; $(MV_DIRS); fi;\
+	      if test -d share/info ; then dir=share/info; $(MV_DIRS) ; fi ; \
+	      rm -fr share ) ) ; \
+	  ( cd $(DESTDIR)${ns_appbindir}libexec && \
+	    ( dir=emacs/*/*/* && $(MV_DIRS); \
+	      rm -fr emacs ) ) ; \
+	  ( cd $(DESTDIR)${ns_appbindir}bin && \
+	    ( rm -f emacs emacs-24* ; \
+	      ln -sf ../libexec/* .) ) ; \
 	else true ; fi
 
 ## FIXME is the emacs-24* bit above really necessary and correct?
--- a/src/ns.mk
+++ b/src/ns.mk
@@ -25,15 +25,15 @@
 ## which appears as a target, is empty on non-NS builds.  Some makes
 ## do not like empty targets, even if they are never used.
 
-${ns_appdir}: ${ns_appsrc}
-	rm -fr ${ns_appdir}
-	mkdir -p ${ns_appdir}
-	( cd ${ns_appsrc} ; tar cfh - . ) | ( cd ${ns_appdir} ; umask 022; tar xf - )
+$(DESTDIR)${ns_appdir}: ${ns_appsrc}
+	rm -fr $(DESTDIR)${ns_appdir}
+	mkdir -p $(DESTDIR)${ns_appdir}
+	( cd ${ns_appsrc} ; tar cfh - . ) | ( cd $(DESTDIR)${ns_appdir} ; umask 022; tar xf - )
 
-${ns_appbindir}Emacs: emacs${EXEEXT}
-	mkdir -p ${ns_appbindir}
-	cp -f emacs${EXEEXT} ${ns_appbindir}Emacs
+$(DESTDIR)${ns_appbindir}Emacs: emacs${EXEEXT}
+	mkdir -p $(DESTDIR)${ns_appbindir}
+	cp -f emacs${EXEEXT} $(DESTDIR)${ns_appbindir}Emacs
 
-ns-app: ${ns_appdir} ${ns_appbindir}Emacs
+ns-app: $(DESTDIR)${ns_appdir} $(DESTDIR)${ns_appbindir}Emacs
 
 ### ns.mk ends here
